src/query/functions/src/scalars/map.rs

```diff
@@ -244,4 +244,41 @@ pub fn register(registry: &mut FunctionRegistry) {
                     .any(|(k, _)| k == key)
             },
         );
+
+    registry.register_2_arg_core::<EmptyMapType, EmptyArrayType, EmptyMapType, _, _>(
+        "map_pick",
+        |_, _, _| FunctionDomain::Full,
+        |_, _, _| Value::Scalar(()),
+    );
+
+    registry.register_2_arg_core::<EmptyMapType, ArrayType<GenericType<0>>, EmptyMapType, _, _>(
+        "map_pick",
+        |_, _, _| FunctionDomain::Full,
+        |_, _, _| Value::Scalar(()),
+    );
+
+    registry.register_passthrough_nullable_2_arg(
```

Comment:
Since `map_pick` has a variable number of arguments, it's better to implement this function using the factory function.
You can refer to these Factory functions as examples

https://github.com/datafuselabs/databend/blob/main/src/query/functions/src/scalars/string_multi_args.rs#L88
https://github.com/datafuselabs/databend/blob/main/src/query/functions/src/scalars/string_multi_args.rs#L247
https://github.com/datafuselabs/databend/blob/main/src/query/functions/src/scalars/geo.rs#L311

Reply:
Currently, I have implemented: `map_pick(map, keys...)`. However, at the same time, is it possible to directly remove the `register_passthrough_nullable_2_arg`, and also need to support ArrayType args in the factory method?